sandy/fy.py

```diff
@@ -350,13 +352,182 @@ def apply_qmatrix(self, zam, energy, decay_data):
+
+        Returns
+        -------
+        `panda.DataFrame`
+            IFY covariance matrix performed by GLS for a given energy and zam.
+
+        Notes
+        -----
+        .. note:: only option `kind='cumulative'` is implemented.
+
+        """
+        # Divide the data type:
+        Vy_calc = self._gls_Vy_calc(zam, e, decay_data, kind, threshold=threshold)
+        # Create the Cov object:
+        Vy_calc = sandy.CategoryCov(Vy_calc)
+        # Fix the S with the correct dimension:
+        S = _gls_setup(decay_data, kind).loc[Vy_calc.data.index, Vy_calc.data.columns]
+        return Vy_calc.gls_update(S, Vy_extra, threshold=threshold).data
+
+    def _gls_y_calc(self, zam, e, decay_data, kind):
+        """
+        “a priori” calculated value for the vector of GLS.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which calculations are to be
+            applied.
+        e : `float`
+            Energy to which calculations are to be applied.
+        decay_data : `DecayData`
+            Container of radioactive nuclide data for several isotopes.
+        kind : `str`
+            Keyword for obtaining sensitivity.
+
+        Raises
+        ------
+        TypeError
+            The kind is not implemented in the method.
+
+        Returns
+        -------
+        y_calc : `pandas.Series`
+            1D calculated output using S.dot(x_prior), e.g. calculated CFY
+
+        """
+        fy_data = self.filter_by('ZAM', zam).filter_by("E", e)
+        if kind == 'cumulative':
+            y_calc = fy_data.apply_qmatrix(zam, e, decay_data).data
+            # Apropiate format
+            y_calc = y_calc[y_calc.MT == 459].set_index('ZAP').FY
+        else:
+            raise TypeError('The kind introduced is not valid')
+        return y_calc
+
+    def _gls_Vy_calc(self, zam, e, decay_data, kind, threshold):
+        """
+        Method to calculate the GLS prior variance according to the model.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which calculations are to be
+            applied.
+        e : `float`
+            Energy to which calculations are to be applied.
+        decay_data : `DecayData`
+            Container of radioactive nuclide data for several isotopes.
+        kind : `str`
+            Keyword for obtaining sensitivity.
+
+        Raises
+        ------
+        TypeError
+            The kind is not implemented in the method.
+
+        Returns
+        -------
+        Vy_calc : `pd.DataFrame`
+            2D calculated output using S.T.dot(Vx_prior.dot(S)).
+
+        """
+        fy_data = self.filter_by('ZAM', zam).filter_by("E", e).data.set_index('ZAP')
+        if kind == 'cumulative':
+            Vx_prior = fy_data.query('MT==454').DFY
+            Vx_prior = sandy.CategoryCov.from_var(Vx_prior)
+            S = decay_data.get_qmatrix().T
```

Comment:
@AitorBengoechea, I wonder if `decay_data.get_qmatrix()` should already return the transpose...because in theory you always need to transpose it if you want to use it...or not?

Reply:
I change the GLS method to be the same of the book Probability, stadistics, and Data Uncertainties in Nuclear Science and technology.